tools/offline_log_viewer/kafka.py

```diff
@@ -32,18 +42,98 @@ def decode(self):
         for batch in self.batches():
             header = batch.header_dict()
             yield header
-            if not self.headers_only:
-                records = []
-                for record in batch:
-                    attrs = header["expanded_attrs"]
-                    is_tx_ctrl = attrs["transactional"] and attrs[
-                        "control_batch"]
-                    record_dict = record.kv_dict()
-                    records.append(record_dict)
-                    if is_tx_ctrl:
-                        record_dict["type"] = self.get_control_record_type(
-                            record.key)
-                    yield record_dict
+            if self.headers_only:
+                continue
+            records = batch
+            if header["type_name"] == "tm_update":
+                records = list(batch)
+                assert len(records) == 1
+            for record in records:
+                attrs = header["expanded_attrs"]
+                is_tx_ctrl = attrs["transactional"] and attrs["control_batch"]
+                record_dict = record.kv_dict()
+                if is_tx_ctrl:
+                    record_dict["type"] = self.get_control_record_type(
+                        record.key)
+                if header["type_name"] == "tm_update":
```

Comment:
I think the otherwise, we should should not ignore errors to catch the divergence as soon as possible to avoid postponing the fix; I'll extract method

Reply:
>I think the otherwise, we should should not ignore errors to catch the divergence as soon as possible to avoid postponing the fix; I'll extract method

yeh, agree we should not ignore any byte parsing by default.